File: src/main/java/frc/robot/utils/NetworkTablesServer.java
package frc.robot.utils;

import edu.wpi.first.networktables.NetworkTableInstance;
import java.net.InetAddress;
import java.util.Arrays;

public class NetworkTablesServer {
    public NetworkTableInstance ntInstance = NetworkTableInstance.getDefault();

    public void setUpNetworkTables() {
        ntInstance.startServer();
        System.out.println("networktables is up");

        try {
            InetAddress ip = InetAddress.getLocalHost();
            System.out.println("IP ADDR" + ip.getHostAddress() + ":5810");
        } catch (Exception e) {
            e.printStackTrace();
        }


        ntInstance.getTable("controls").getEntry("moveForward").setBoolean(false);
        ntInstance.getTable("controls").getEntry("moveBackward").setBoolean(false);
        ntInstance.getTable("controls").getEntry("moveLeft").setBoolean(false);
        ntInstance.getTable("controls").getEntry("moveRight").setBoolean(false);

        System.out.println(ntInstance.getTable("controls").getValue("moveRight"));

        // simple loop
        while (true) {
            try {
                Thread.sleep(1000);
//                System.out.println(Arrays.toString(ntInstance.getConnections()));
            } catch (InterruptedException ex) {
                System.out.println("Interrupted!");
                return;
            }
        }
    }
}

File: src/main/java/frc/robot/utils/NetworkTablesServer.kt
package frc.robot.utils

import edu.wpi.first.networktables.BooleanSubscriber
import edu.wpi.first.networktables.DoubleSubscriber
import edu.wpi.first.networktables.NetworkTable
import edu.wpi.first.networktables.NetworkTableInstance
import java.net.InetAddress

class NetworkTablesServer {
    private var ntInstance: NetworkTableInstance = NetworkTableInstance.getDefault()

    private val controls: NetworkTable = ntInstance.getTable("controls")
    private val dir: DoubleSubscriber
    private val rot: DoubleSubscriber
    private val robotEnabled: BooleanSubscriber

    // primitives
    var dirVal: Double = 0.0
        private set
    var rotVal: Double = 0.0
        private set
    var robotEnabledVal: Boolean = false
        private set

    private fun init() {
        try {
            val ip = InetAddress.getLocalHost()
            println("IP ADDR NT4 " + ip.hostAddress + ":5810")
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    init {
        init()

        dir = controls.getDoubleTopic("dir").subscribe(0.0)
        rot = controls.getDoubleTopic("rot").subscribe(0.0)
        robotEnabled = controls.getBooleanTopic("robotEnabled").subscribe(false)
    }

    fun updateValuesPeriodic() {
        dirVal = dir.get()
        rotVal = rot.get()
        robotEnabledVal = robotEnabled.get()

//        println("Dir: $dirVal Rot: $rotVal")
    }

    fun close() {
        dir.close()
        rot.close()
    }
}
